opensearch-operator/pkg/helpers/helpers.go

```diff
@@ -482,6 +482,12 @@ func ReplicaHostName(currentSts appsv1.StatefulSet, repNum int32) string {
 }
 
 func WorkingPodForRollingRestart(k8sClient k8s.K8sClient, sts *appsv1.StatefulSet) (string, error) {
+	// Skip rolling restart if revisions are the same (no actual spec changes)
+	// This prevents unnecessary rolling restarts during node draining/pod recreation
+	if sts.Status.UpdateRevision != "" && sts.Status.CurrentRevision == sts.Status.UpdateRevision {
+		return "", errors.New("no rolling restart needed - StatefulSet revisions match")
+	}
```

Comment:
This error is bubbled up till the reconcile func. It means this error triggers reconcile requeue.
I think we don't need to requeue.
Now this func doesn't have response case with nil error and empty pod name. Therefore, the caller parts of this func just check if `err==nil` and not check working pod name.

I'd rather set the condition check for working pod name separately. Then we can return nil error for this non-revision change case with empty working pod name.
@synhershko what do you think?